File: addons/tr03112/src/jvmMain/java/org/openecard/sal/protocol/eac/gui/PinState.java
/****************************************************************************
 * Copyright (C) 2019-2020 ecsec GmbH.
 * All rights reserved.
 * Contact: ecsec GmbH (info@ecsec.de)
 *
 * This file may be used in accordance with the terms and conditions
 * contained in a signed written agreement between you and ecsec GmbH.
 *
 ***************************************************************************/

package org.openecard.sal.protocol.eac.gui;

import org.openecard.common.ifd.PacePinStatus;


/**
 *
 * @author Tobias Wich
 */
public class PinState {

    private PacePinStatus state;

    public PinState() {
	state = PacePinStatus.RC3;
    }

    public void update(PacePinStatus status) {
	if (status == null) {
	    status = PacePinStatus.UNKNOWN;
	}
	state = status;
    }

    public PacePinStatus getState() {
	return state;
    }

    public int getAttempts() {
	switch (state) {
	    case RC3:
		return 3;
	    case RC2:
		return 2;
	    case RC1:
		return 1;
	    default:
		return 0;
	}
    }

    public boolean isRequestCan() {
	return state == PacePinStatus.RC1;
    }

    public boolean isBlocked() {
	return state == PacePinStatus.BLOCKED;
    }

    public boolean isDeactivated() {
	return state == PacePinStatus.DEACTIVATED;
    }

    public boolean isOperational() {
	return ! isBlocked() && ! isDeactivated();
    }

    public boolean isUnknown() {
	return state == PacePinStatus.UNKNOWN;
    }

}

File: addons/tr03112/src/jvmMain/kotlin/org/openecard/sal/protocol/eac/gui/PinState.kt
/****************************************************************************
 * Copyright (C) 2019-2020 ecsec GmbH.
 * All rights reserved.
 * Contact: ecsec GmbH (info@ecsec.de)
 *
 * This file may be used in accordance with the terms and conditions
 * contained in a signed written agreement between you and ecsec GmbH.
 *
 */
package org.openecard.sal.protocol.eac.gui

import org.openecard.common.ifd.PacePinStatus

/**
 *
 * @author Tobias Wich
 */
class PinState {
	private var state: PacePinStatus = PacePinStatus.RC3

	fun update(status: PacePinStatus?) {
		var status = status
		if (status == null) {
			status = PacePinStatus.UNKNOWN
		}
		state = status
	}

	fun getState(): PacePinStatus = state

	val attempts: Int
		get() {
			return when (state) {
				PacePinStatus.RC3 -> 3
				PacePinStatus.RC2 -> 2
				PacePinStatus.RC1 -> 1
				else -> 0
			}
		}

	val isRequestCan: Boolean
		get() = state == PacePinStatus.RC1

	val isBlocked: Boolean
		get() = state == PacePinStatus.BLOCKED

	val isDeactivated: Boolean
		get() = state == PacePinStatus.DEACTIVATED

	val isOperational: Boolean
		get() = !this.isBlocked && !this.isDeactivated

	val isUnknown: Boolean
		get() = state == PacePinStatus.UNKNOWN
}
